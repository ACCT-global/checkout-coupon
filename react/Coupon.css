.inputForm :global(.vtex-input) input {
  min-width: 0;
}

.couponWrapper {}

.tagWrapper {}

.tagText {}