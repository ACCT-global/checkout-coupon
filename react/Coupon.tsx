import React, { useState }  from 'react'
import { Button, Input} from 'vtex.styleguide'
import { compose, graphql } from 'react-apollo'
// import { Spinner } from 'vtex.styleguide'

import * as Coupons from './graphql/coupons.graphql'

import styles from './styles.css'

const Coupon: StorefrontFunctionComponent<CouponProps> = ({ }) => {

  const [isShowingPromoButton, setIsShowingPromoButton ] = useState(true);
  const toggle = () => setIsShowingPromoButton(!isShowingPromoButton)

  return (
    <div className={`${styles.container} flex flex-column pv6 ph4`}>
      <div className="mb4 mr4">
        <div className="ma0 dib">
          <h3 className="t-heading-3">Promotions</h3>

          {isShowingPromoButton &&
          <Button variation="tertiary" onClick={toggle} collapseLeft>APPLY PROMO CODE</Button>}


          {!isShowingPromoButton && 
            <div className="flex">
              <Input
                placeholder="Type here"
                dataAttributes={{ 'hj-white-list': true, test: 'string' }}
                label="Promo Code"
              />

              <Button variation="tertiary" size="regular" onClick={toggle}>APPLY</Button>
            </div>
          }

          
        </div>
      </div>
    
    </div>
  )
}

interface CouponProps {
  title?: string,
  CouponsQuery: any
}

Coupon.schema = {
  title: 'editor.base-store-component.title',
  description: 'editor.base-store-component.description',
  type: 'object',
  properties: {
    title: {
      title: 'editor.base-store-component.title.title',
      description: 'editor.base-store-component.title.description',
      type: 'string',
      default: null,
    },
  },
}

export default compose(
  graphql(Coupons.default, {
    options: { ssr: false}  
  })
) (Coupon)