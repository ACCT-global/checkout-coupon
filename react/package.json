{
  "scripts": {
    "test": "vtex-test-tools test",
    "lint": "tsc --noEmit && eslint --ext ts,tsx ."
  },
  "dependencies": {
    "@vtex/css-handles": "^1.0.0",
    "apollo-boost": "^0.4.3",
    "classnames": "^2.2.6",
    "graphql": "^14.4.2",
    "graphql-tag": "^2.10.1",
    "ramda": "^0.26.1",
    "react": "^16.8.3",
    "react-apollo": "^2.5.8",
    "react-intl": "^2.7.2"
  },
  "devDependencies": {
    "@types/classnames": "^2.2.7",
    "@types/graphql": "^14.2.0",
    "@types/jest": "^24.0.11",
    "@types/node": "^11.13.0",
    "@types/prop-types": "^15.7.0",
    "@types/ramda": "^0.26.5",
    "@types/react": "^16.8.10",
    "@types/react-intl": "^2.3.17",
    "@vtex/styleguide": "^9.91.3",
    "@vtex/test-tools": "^0.2.0",
    "apollo-client": "^2.6.3",
    "react-dom": "^16.8.6",
    "typescript": "3.7.3",
    "vtex.render-runtime": "https://browserslist--storecomponents.myvtex.com/_v/private/typings/linked/v1/vtex.render-runtime@8.111.0+build1594327796/public/@types/vtex.render-runtime",
    "vtex.styleguide": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.styleguide@9.124.2/public/@types/vtex.styleguide"
  },
  "version": "0.10.1"
}
