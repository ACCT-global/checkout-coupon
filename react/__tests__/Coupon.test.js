import expect from 'expect'
import { render } from '@vtex/test-tools/react'
import React from 'react'
import Coupon from '../Coupon.tsx'

describe('CouponComponent', () => {
  const renderComponent = customProps => {
    const wrapper = render(<Coupon></Coupon>)
    return wrapper
  }

  it('should be rendered', () => {
    const component = renderComponent()
    expect(component).toBeDefined()
  })

  it('should match the snapshot', () => {
    const component = renderComponent()
    expect(component).toMatchSnapshot()
  })
})
