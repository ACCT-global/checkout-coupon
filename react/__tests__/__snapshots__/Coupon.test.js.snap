// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponComponent should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container flex flex-column pv6 ph4"
      >
        <div
          class="mb4 mr4"
        >
          <div
            class="ma0 dib"
          >
            <h3
              class="t-heading-3"
            >
              Promotions
            </h3>
            <button
              class="vtex-button bw1 ba fw5 v-mid relative pa0 br2 min-h-regular t-action nl6 bg-transparent b--transparent c-action-primary hover-b--transparent pointer "
              tabindex="0"
              type="button"
            >
              <div
                class="flex items-center justify-center h-100 pv2 ph6 nl1 "
              >
                APPLY PROMO CODE
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container flex flex-column pv6 ph4"
    >
      <div
        class="mb4 mr4"
      >
        <div
          class="ma0 dib"
        >
          <h3
            class="t-heading-3"
          >
            Promotions
          </h3>
          <button
            class="vtex-button bw1 ba fw5 v-mid relative pa0 br2 min-h-regular t-action nl6 bg-transparent b--transparent c-action-primary hover-b--transparent pointer "
            tabindex="0"
            type="button"
          >
            <div
              class="flex items-center justify-center h-100 pv2 ph6 nl1 "
            >
              APPLY PROMO CODE
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
